### PR TITLE
Add AppBar.forceMaterialTransparency (#101248)

### LINKED ISSUE #101248: Cannot touch widget behind a (transparent & unelevated) AppBar when `extendBodyBehindAppBar` is `true`
## Steps to Reproduce

1. Execute `flutter run` on the code sample 
2. Attempt to tap the button which is behind the transparent & unelevated AppBar

**Expected results:** 
Button accepts tap

**Actual results:** 
Button does not accept tap

**Reason:**
AppBar uses a Material widget (app_bar.dart line 1060) with an unspecified MaterialType, which defaults to MaterialType.canvas. Material widgets absorb touches unless the type is MaterialType.transparency.

**Suggestion:**
If the AppBar's elevation is zero and background color is transparent, then set the Material within the AppBar to type = MaterialType.transparency. This allows touches to pass through instead of being absorbed. 

I will create a PR for this, but am open to any other strategies (e.g. an explicit parameter on AppBar).

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData.dark(),
      home: Scaffold(
        extendBodyBehindAppBar: true,
        appBar: AppBar(
          title: Text('I am an app bar'),
          backgroundColor: Color(0x00000000),
          elevation: 0,
        ),
        body: Align(
            alignment: Alignment.topCenter,
            child: ElevatedButton(
                onPressed: () => print('tap!'), child: Text('tap me'))),
      ),
    );
  }
}
```

</details>

### PR DESCRIPTION
**Pull request by monkeyswarm**
Allows gestures to reach widgets beneath the AppBar (when Scaffold.extendBodyBehindAppBar is true).

*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

Fixes https://github.com/flutter/flutter/issues/101248
Previous discussion at previous strategy https://github.com/flutter/flutter/pull/102055 suggested using a new flag on AppBar, which this PR does, instead of attempting to derive intended MaterialType from color/elevation. Naming convention open for discussion.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing NOTE there are Pixel tests failing at head, unrelated to this PR.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by HansMuller — COMMENTED** [review]
This LGTM, just some code formatting nits.

File: packages/flutter/lib/src/material/app_bar.dart [hunk 1]
```diff
@@ -789,6 +790,21 @@ class AppBar extends StatefulWidget implements PreferredSizeWidget {
   ///  * [SystemChrome.setSystemUIOverlayStyle]
   final SystemUiOverlayStyle? systemOverlayStyle;
 
+  /// {@template flutter.material.appbar.forceMaterialTransparency}
```
[the remark this answers is not in the document]
Comment (monkeyswarm) [reply packages/flutter/lib/src/material/app_bar.dart:793]: Good question; I used it to copy the pattern of all other AppBar fields, but I neglected to actually use the template in SliverAppBar. Is it expected/required that SliverAppBar have all the same fields as AppBar?

[the remark this answers is not in the document]
Comment (HansMuller) [reply packages/flutter/lib/src/material/app_bar.dart:793]: Yes! Odd way to discover that oversight, but still helpful.

File: packages/flutter/test/material/app_bar_test.dart [hunk 2]
```diff
@@ -3760,4 +3760,78 @@ void main() {
...
+        home: Scaffold(
+          extendBodyBehindAppBar: true,
+          appBar: AppBar(
+            forceMaterialTransparency: forceMaterialTransparency,
+            elevation: 3,
+            backgroundColor: Colors.red,
+            title: const Text('AppBar'),
+          ),
+          body: Align(
+              alignment: Alignment.topCenter,
```
Comment (HansMuller) [inline packages/flutter/test/material/app_bar_test.dart:3787]: two space indent here

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**